managed/src/main/java/com/yugabyte/yw/common/KubernetesManager.java

```diff
@@ -105,6 +107,24 @@ public ShellResponse getServiceIPs(Map<String, String> config,
     return execCommand(config, commandList);
   }
 
+  public ShellResponse getNodeInfos(Map<String, String> config) {
+    List<String> commandList = ImmutableList.of("kubectl", "get", "nodes", "-o", "json");
+    return execCommand(config, commandList);
+  }
+
+  // TODO(bhavin192): disable the logging of stdout of this command if
+  // possibile, as it just leaks the secret content in the logs at
+  // debug level.
+  public ShellResponse getSecret(Map<String, String> config, String secretName, String namespace) {
```

Comment:
same comment: let's return the jsonnode directly to the caller (or String) instead of a ShellResponse object. we can throw runtime exceptions or YWServiceException on failure to let the failure bubble up to the API level directly.